File: app/src/main/java/cn/readsense/mystudy/MainActivity.java
package cn.readsense.mystudy;

import android.content.Intent;
import android.icu.util.Output;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.view.Menu;
import android.view.MenuItem;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.RandomAccessFile;
import java.io.Writer;
import java.util.ArrayList;
import java.util.Collections;
import java.util.PriorityQueue;
import java.util.Stack;
import java.util.Vector;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });
        test();
    }

    private void test() {
        Intent intent = new Intent();
        Bundle bundle = new Bundle();
        intent.putExtra("ddd",111);
        intent.putExtras(bundle);
        Stack<String> stack = new Stack<>();
        stack.push(null);
        stack.push(null);
        stack.pop();

        try {
            RandomAccessFile randomAccessFile = new RandomAccessFile("","");
            randomAccessFile.getChannel();
            File file = new File("");
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: skinapp/src/main/java/cn/readsense/skinapp/MainActivity.kt
package cn.readsense.skinapp

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import cn.readsense.skinapp.fragment.MusicFragment
import cn.readsense.skinapp.fragment.RadioFragment
import cn.readsense.skinapp.fragment.VideoFragment
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //控件的收集在这里完成
        setContentView(R.layout.activity_main)
        val list: MutableList<Fragment> = ArrayList<Fragment>()
        list.add(MusicFragment())
        list.add(VideoFragment())
        list.add(RadioFragment())
        val listTitle: MutableList<String> = ArrayList()
        listTitle.add("音乐")
        listTitle.add("视频")
        listTitle.add("电台")
        val myFragmentPagerAdapter = MyFragmentPagerAdapter(supportFragmentManager, list, listTitle)
        viewPager.adapter = myFragmentPagerAdapter
        tabLayout.setupWithViewPager(viewPager)
    }

    /**
     * 进入换肤
     *
     * @param view
     */
    fun skinSelect(view: View) {
        startActivity(Intent(this, SkinActivity::class.java))
    }
}